cpp/src/arrow/util/decimal_test.cc

```diff
@@ -590,13 +597,62 @@ TEST(TestDecimalFromRealFloat, LargeValues) {
   // Test the entire float range
   for (int32_t scale = -38; scale <= 38; ++scale) {
     float real = std::pow(10.0f, static_cast<float>(scale));
-    CheckDecimalFromRealIntegerString(real, 1, -scale, "1");
+    CheckDecimalFromRealIntegerString<Decimal128>(real, 1, -scale, "1");
```

Comment:
Two things:
- This apparently fails on Windows MinGW32 due to precision issues.
- This didn't fail before - something changed. Possibly I do have to figure out how to use ASSERT_FLOAT_EQ/ASSERT_DOUBLE_EQ as appropriate.